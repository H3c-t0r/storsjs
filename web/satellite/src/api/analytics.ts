// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

import { HttpClient } from '@/utils/httpClient';

/**
 * AnalyticsHttpApi is a console Analytics API.
 * Exposes all analytics-related functionality
 */
export class AnalyticsHttpApi {
    private readonly http: HttpClient = new HttpClient();
    private readonly ROOT_PATH: string = '/api/v0/analytics';

    /**
     * Used to get authentication token.
     *
     * @param eventName - name of the event
     * @throws Error
     */
    public async eventTriggered(eventName: string): Promise<void> {
        const path = `${this.ROOT_PATH}/event`;
        const body = {
            eventName: eventName,
        };
        const response = await this.http.post(path, JSON.stringify(body));
        if (response.ok) {
            return;
        }

        throw new Error('Can not track event');
    }
}