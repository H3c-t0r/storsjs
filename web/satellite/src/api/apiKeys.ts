// Copyright (C) 2018 Storj Labs, Inc.
// See LICENSE for copying information.

import apollo from '@/utils/apolloManager';
import gql from 'graphql-tag';

export async function fetchAPIKeys(projectID: string) {
    let result: RequestResponse<any[]> = {
        errorMessage: '',
        isSuccess: false,
        data: []
    };

    try {
        let response: any = await apollo.query({
            query: gql(
                `query {
                    project(
                        id: "${projectID}",
                    ) {
                        apiKeys {
                            id,
                            name,
                            createdAt
                        }
                    }
                }`
            ),
            fetchPolicy: 'no-cache'
        });

        if (response.errors) {
            result.errorMessage = response.errors[0].message;
        } else {
            result.isSuccess = true;
            result.data = response.data.project.apiKeys;
        }
    } catch (e) {
        result.errorMessage = e.message;
    }

    return result;
}

export async function createAPIKey(projectID: string, name: string) {
    let result: RequestResponse<any> = {
        errorMessage: '',
        isSuccess: false,
        data: null
    };

    try {
        let response: any = await apollo.mutate({
            mutation: gql(
                `mutation {
                    createAPIKey(
                        projectID: "${projectID}",
                        name: "${name}"
                    ) {
                        key,
                        keyInfo {
                            id,
                            name,
                            createdAt
                        }
                    }
                }`
            ),
            fetchPolicy: 'no-cache'
        });

        if (response.errors) {
            result.errorMessage = response.errors[0].message;
        } else {
            console.log('response', response)
            result.isSuccess = true;
            result.data = {
                key: response.data.createAPIKey.key,
                keyInfo: response.data.createAPIKey.keyInfo
            }
        }
    } catch (e) {
        result.errorMessage = e.message;
    }

    return result;
}

export async function deleteAPIKey(id: string) {
    let result: RequestResponse<any> = {
        errorMessage: '',
        isSuccess: false,
        data: null
    };

    try {
        let response: any = await apollo.mutate({
            mutation: gql(
                `mutation {
                    deleteAPIKey(id: "${id}") {
                        id
                    }
                }`
            ),
            fetchPolicy: 'no-cache'
        });

        if (response.errors) {
            result.errorMessage = response.errors[0].message;
        } else {
            result.isSuccess = true;
            result.data = response.data.deleteAPIKey;
        }
    } catch (e) {
        result.errorMessage = e.message;
    }

    return result;
}