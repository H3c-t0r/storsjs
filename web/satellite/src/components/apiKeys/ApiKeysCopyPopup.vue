// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="save-api-popup" v-if="isPopupShown">
        <h2 class="save-api-popup__title">Save Your API Key! It Will Appear Only Once.</h2>
        <div class="save-api-popup__copy-area">
            <div class="save-api-popup__copy-area__key-area">
                <p class="save-api-popup__copy-area__key-area__key">{{apiKeySecret}}</p>
            </div>
            <div class="copy-button" v-clipboard="apiKeySecret" @click="onCopyClick" v-if="!isCopiedButtonShown">
                <svg width="22" height="22" viewBox="0 0 22 22" fill="none" xmlns="http://www.w3.org/2000/svg">
                    <path d="M13.3763 7.7002H3.34383C2.46383 7.7002 1.76001 8.40402 1.76001 9.28402V19.3378C1.76001 20.1954 2.46383 20.9216 3.34383 20.9216H13.3976C14.2553 20.9216 14.9814 20.2178 14.9814 19.3378L14.9823 9.28402C14.96 8.40402 14.2561 7.7002 13.3761 7.7002H13.3763ZM13.6401 19.3164C13.6401 19.4488 13.5301 19.5588 13.3977 19.5588L3.34397 19.5579C3.21162 19.5579 3.10161 19.4479 3.10161 19.3156L3.10247 9.284C3.10247 9.15165 3.21247 9.04164 3.34483 9.04164H13.3986C13.5309 9.04164 13.641 9.15164 13.641 9.284L13.6401 19.3164Z" fill="white"/>
                    <path d="M18.6563 1.09974H8.62386C7.74386 1.09974 7.04004 1.80356 7.04004 2.68356V6.37978H8.36004V2.68356C8.36004 2.55122 8.47004 2.44121 8.60239 2.44121H18.6562C18.7885 2.44121 18.8985 2.55121 18.8985 2.68356V12.7373C18.8985 12.8697 18.7885 12.9797 18.6562 12.9797H16.2799V14.2997H18.6562C19.5138 14.2997 20.24 13.5959 20.24 12.7159V2.68343C20.24 1.80343 19.5362 1.09961 18.6562 1.09961L18.6563 1.09974Z" fill="#354049"/>
                    <rect x="2.93335" y="8.7998" width="11.7333" height="11" fill="white"/>
                    <rect x="7.1001" y="1.2334" width="12.9333" height="12.9333" rx="1.5" fill="white" stroke="#2683FF"/>
                </svg>
                <p class="copy-button__label">Copy</p>
            </div>
            <div class="copied-button" v-if="isCopiedButtonShown">
                <svg width="22" height="22" viewBox="0 0 22 22" fill="none" xmlns="http://www.w3.org/2000/svg">
                    <path d="M13.3763 7.7002H3.34383C2.46383 7.7002 1.76001 8.40402 1.76001 9.28402V19.3378C1.76001 20.1954 2.46383 20.9216 3.34383 20.9216H13.3976C14.2553 20.9216 14.9814 20.2178 14.9814 19.3378L14.9823 9.28402C14.96 8.40402 14.2561 7.7002 13.3761 7.7002H13.3763ZM13.6401 19.3164C13.6401 19.4488 13.5301 19.5588 13.3977 19.5588L3.34397 19.5579C3.21162 19.5579 3.10161 19.4479 3.10161 19.3156L3.10247 9.284C3.10247 9.15165 3.21247 9.04164 3.34483 9.04164H13.3986C13.5309 9.04164 13.641 9.15164 13.641 9.284L13.6401 19.3164Z" fill="white"/>
                    <path d="M18.6563 1.09974H8.62386C7.74386 1.09974 7.04004 1.80356 7.04004 2.68356V6.37978H8.36004V2.68356C8.36004 2.55122 8.47004 2.44121 8.60239 2.44121H18.6562C18.7885 2.44121 18.8985 2.55121 18.8985 2.68356V12.7373C18.8985 12.8697 18.7885 12.9797 18.6562 12.9797H16.2799V14.2997H18.6562C19.5138 14.2997 20.24 13.5959 20.24 12.7159V2.68343C20.24 1.80343 19.5362 1.09961 18.6562 1.09961L18.6563 1.09974Z" fill="#354049"/>
                    <rect x="2.93335" y="8.7998" width="11.7333" height="11" fill="white"/>
                    <rect x="7.1001" y="1.2334" width="12.9333" height="12.9333" rx="1.5" fill="white" stroke="#2683FF"/>
                </svg>
                <p class="copied-button__label">Copied</p>
            </div>
        </div>
        <div class="save-api-popup__close-cross-container" @click="onCloseClick">
            <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
                <path class="close-cross-svg-path" d="M15.7071 1.70711C16.0976 1.31658 16.0976 0.683417 15.7071 0.292893C15.3166 -0.0976311 14.6834 -0.0976311 14.2929 0.292893L15.7071 1.70711ZM0.292893 14.2929C-0.0976311 14.6834 -0.0976311 15.3166 0.292893 15.7071C0.683417 16.0976 1.31658 16.0976 1.70711 15.7071L0.292893 14.2929ZM1.70711 0.292893C1.31658 -0.0976311 0.683417 -0.0976311 0.292893 0.292893C-0.0976311 0.683417 -0.0976311 1.31658 0.292893 1.70711L1.70711 0.292893ZM14.2929 15.7071C14.6834 16.0976 15.3166 16.0976 15.7071 15.7071C16.0976 15.3166 16.0976 14.6834 15.7071 14.2929L14.2929 15.7071ZM14.2929 0.292893L0.292893 14.2929L1.70711 15.7071L15.7071 1.70711L14.2929 0.292893ZM0.292893 1.70711L14.2929 15.7071L15.7071 14.2929L1.70711 0.292893L0.292893 1.70711Z" fill="#384B65"/>
            </svg>
        </div>
        <div class="blur-content"></div>
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import HeaderlessInput from '@/components/common/HeaderlessInput.vue';

import { NOTIFICATION_ACTIONS } from '@/utils/constants/actionNames';

@Component({
    components: {
        HeaderlessInput,
    },
})
export default class ApiKeysCopyPopup extends Vue {
    @Prop({default: false})
    private readonly isPopupShown: boolean;
    @Prop({default: ''})
    private readonly apiKeySecret: string;

    public isCopiedButtonShown: boolean = false;

    public onCloseClick(): void {
        this.isCopiedButtonShown = false;
        this.$emit('closePopup');
    }

    public onCopyClick(): void {
        this.$store.dispatch(NOTIFICATION_ACTIONS.SUCCESS, 'Key saved to clipboard');
        this.isCopiedButtonShown = true;
    }
}
</script>

<style scoped lang="scss">
    .save-api-popup {
        padding: 32px 40px 60px 40px;
        background-color: #FFFFFF;
        border-radius: 24px;
        margin-top: 29px;
        max-width: 94.8%;
        height: auto;
        position: relative;
        font-family: 'font_regular';

        &__title {
            font-family: 'font_bold';
            font-size: 24px;
            line-height: 29px;
            margin-bottom: 26px;
        }

        &__copy-area {
            display: flex;
            align-items: center;
            justify-content: space-between;
            background-color: #F5F6FA;
            padding: 29px 32px 29px 24px;
            border-radius: 12px;
            position: relative;

            &__key-area {

                &__key {
                    margin: 0;
                    font-size: 16px;
                    line-height: 21px;
                    word-break: break-all;
                }
            }

            .copy-button,
            .copied-button {
                display: flex;
                background-color: #2683FF;
                padding: 13px 36px;
                cursor: pointer;
                align-items: center;
                justify-content: space-between;
                color: #FFFFFF;
                border: 1px solid #2683FF;
                box-sizing: border-box;
                border-radius: 8px;
                font-size: 14px;
                font-family: 'font_bold';
                margin-left: 10px;

                &__label {
                    margin: 0;
                }

                &:hover {
                    background-color: #196CDA;
                }
            }

            .copied-button {
                padding: 13px 28.5px;
                background-color: #196CDA;
                cursor: default;
            }
        }

        &__close-cross-container {
            display: flex;
            justify-content: center;
            align-items: center;
            position: absolute;
            right: 29px;
            top: 29px;
            height: 24px;
            width: 24px;
            cursor: pointer;

            &:hover .close-cross-svg-path {
                fill: #2683FF;
            }
        }

        .blur-content {
            position: absolute;
            top: 100%;
            left: 0;
            background-color: #F5F6FA;
            width: 100%;
            height: 70vh;
            z-index: 100;
            opacity: 0.3;
        }
    }
</style>
