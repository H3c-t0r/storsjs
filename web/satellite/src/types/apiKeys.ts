// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Exposes all apiKey-related functionality
 */
export interface ApiKeysApi {
    /**
     * Fetch apiKey
     *
     * @returns ApiKey
     * @throws Error
     */
    get(projectId: string): Promise<ApiKey[]>;
    create(projectId: string, name: string): Promise<ApiKey>;
    delete(ids: string[]): Promise<null>;
}

/**
 * ApiKey class holds info for ApiKey entity.
 */
export class ApiKey {
    public id: string;
    public secret: string;
    public name: string;
    public createdAt: string;
    public isSelected: boolean = false;

    constructor(id: string, name: string, createdAt: string, secret: string) {
        this.id = id || '';
        this.name = name || '';
        this.createdAt = createdAt || '';
        this.secret = secret || '';

        this.isSelected = false;
    }

    public formattedName(): string {
        let name = this.name;

        if (name.length < 12) {
            return name;
        }

        return name.slice(0, 12) + '...';
    }

    public getDate(): string {
        if (!this.createdAt) {
            return '';
        }

        return new Date(this.createdAt).toLocaleDateString();
    }
}
