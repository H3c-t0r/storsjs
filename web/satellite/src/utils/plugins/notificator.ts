// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { store as globalStore } from '@/store';
import { NOTIFICATION_ACTIONS } from '@/utils/constants/actionNames';

interface Dispatcher {
    dispatch(key:string, message:string)
}

/**
 * Exposes UI notifications functionality.
 */
export class Notificator {
    public constructor(private store: Dispatcher) {}

    public async success(message: string): Promise<void> {
        await this.store.dispatch(NOTIFICATION_ACTIONS.SUCCESS, message);
    }

    public async error(message: string): Promise<void> {
        await this.store.dispatch(NOTIFICATION_ACTIONS.ERROR, message);
    }

    public async notify(message: string): Promise<void> {
        await this.store.dispatch(NOTIFICATION_ACTIONS.NOTIFY, message);
    }

    public async warning(message: string): Promise<void> {
        await this.store.dispatch(NOTIFICATION_ACTIONS.WARNING, message);
    }
}

/**
 * Registers plugin in Vue instance.
 */
export class NotificatorPlugin {
    public install(localVue: { prototype: { $notify: Notificator } }): void {
        localVue.prototype.$notify = new Notificator(globalStore);
    }
}
