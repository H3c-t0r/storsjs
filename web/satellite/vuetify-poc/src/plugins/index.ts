// Copyright (C) 2023 Storj Labs, Inc.
// See LICENSE for copying information.

import { createPinia, setActivePinia } from 'pinia';
import { App } from 'vue';

import { router } from '../router';

import { loadFonts } from './webfontloader';
import vuetify from './vuetify';

const pinia = createPinia();
setActivePinia(pinia);

export function registerPlugins(app: App<Element>): void {
    loadFonts();
    app
        .use(vuetify)
        .use(router)
        .use(pinia);
}
