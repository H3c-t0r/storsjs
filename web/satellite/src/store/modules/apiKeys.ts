// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { StoreModule } from '@/store';
import { ApiKey, ApiKeyCursor, ApiKeyOrderBy, ApiKeysApi, ApiKeysPage } from '@/types/apiKeys';
import { SortDirection } from '@/types/common';

export const API_KEYS_MUTATIONS = {
    SET_PAGE: 'setApiKeys',
    TOGGLE_SELECTION: 'toggleApiKeysSelection',
    CLEAR_SELECTION: 'clearApiKeysSelection',
    CLEAR: 'clearApiKeys',
    CHANGE_SORT_ORDER: 'changeApiKeysSortOrder',
    CHANGE_SORT_ORDER_DIRECTION: 'changeApiKeysSortOrderDirection',
    SET_SEARCH_QUERY: 'setApiKeysSearchQuery',
    SET_PAGE_NUMBER: 'setApiKeysPage',
};

const {
    SET_PAGE,
    TOGGLE_SELECTION,
    CLEAR_SELECTION,
    CLEAR,
    CHANGE_SORT_ORDER,
    CHANGE_SORT_ORDER_DIRECTION,
    SET_SEARCH_QUERY,
    SET_PAGE_NUMBER,
} = API_KEYS_MUTATIONS;

class ApiKeysState {
    public cursor: ApiKeyCursor = new ApiKeyCursor();
    public page: ApiKeysPage = new ApiKeysPage();
}

/**
 * creates apiKeys module with all dependencies
 *
 * @param api - apiKeys api
 */
export function makeApiKeysModule(api: ApiKeysApi): StoreModule<ApiKeysState> {
    return {
        state: new ApiKeysState(),
        mutations: {
            [SET_PAGE](state: ApiKeysState, page: ApiKeysPage) {
                state.page = page;
            },
            [SET_PAGE_NUMBER](state: ApiKeysState, pageNumber: number) {
                state.cursor.page = pageNumber;
            },
            [SET_SEARCH_QUERY](state: ApiKeysState, search: string) {
                state.cursor.search = search;
            },
            [CHANGE_SORT_ORDER](state: ApiKeysState, order: ApiKeyOrderBy) {
                state.cursor.order = order;
            },
            [CHANGE_SORT_ORDER_DIRECTION](state: ApiKeysState, direction: SortDirection) {
                state.cursor.orderDirection = direction;
            },
            [TOGGLE_SELECTION](state: ApiKeysState, apiKeyID: string) {
                state.page.apiKeys = state.page.apiKeys.map((apiKey: ApiKey) => {
                    if (apiKey.id === apiKeyID) {
                        apiKey.isSelected = !apiKey.isSelected;
                    }

                    return apiKey;
                });
            },
            [CLEAR_SELECTION](state: ApiKeysState) {
                state.page.apiKeys = state.page.apiKeys.map((apiKey: ApiKey) => {
                    apiKey.isSelected = false;

                    return apiKey;
                });
            },
            [CLEAR](state: ApiKeysState) {
                state.cursor = new ApiKeyCursor();
                state.page = new ApiKeysPage();
            },
        },
        actions: {
            fetchApiKeys: async function ({commit, rootGetters, state}, pageNumber: number): Promise<ApiKeysPage> {
                const projectId = rootGetters.selectedProject.id;
                commit(SET_PAGE_NUMBER, pageNumber);

                const apiKeys = await api.get(projectId, state.cursor);
                commit(SET_PAGE, apiKeys);

                return apiKeys;
            },
            createApiKey: async function ({commit, rootGetters}: any, name: string): Promise<ApiKey> {
                const projectId = rootGetters.selectedProject.id;

                const apiKey = await api.create(projectId, name);

                return apiKey;
            },
            deleteApiKey: async function({commit}: any, ids: string[]): Promise<void> {
                return await api.delete(ids);
            },
            setApiKeysSearchQuery: function ({commit}, search: string) {
                commit(SET_SEARCH_QUERY, search);
            },
            setApiKeysSortingBy: function ({commit}, order: ApiKeyOrderBy) {
                commit(CHANGE_SORT_ORDER, order);
            },
            setApiKeysSortingDirection: function ({commit}, direction: SortDirection) {
                commit(CHANGE_SORT_ORDER_DIRECTION, direction);
            },
            toggleApiKeySelection: function ({commit}, apiKeyID: string): void {
                commit(TOGGLE_SELECTION, apiKeyID);
            },
            clearApiKeySelection: function ({commit}): void {
                commit(CLEAR_SELECTION);
            },
            clearApiKeys: function ({commit}): void {
                commit(CLEAR);
            },
        },
        getters: {
            selectedApiKeys: (state: ApiKeysState) => state.page.apiKeys.filter((key: ApiKey) => key.isSelected),
            apiKeys: function (state: ApiKeysState): ApiKey[] {
                return state.page.apiKeys;
            },
        },
    };
}
