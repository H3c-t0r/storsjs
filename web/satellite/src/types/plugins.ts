// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import Vue from 'vue';

import { Notificator } from '@/utils/plugins/notificator';

declare module 'vue/types/vue' {
    interface Vue {
        $segment: any; // define real typings here if you want
        $notify: Notificator;
    }
}
