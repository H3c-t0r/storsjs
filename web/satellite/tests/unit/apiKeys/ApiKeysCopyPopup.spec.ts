// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import Vuex from 'vuex';

import ApiKeysCopyPopup from '@/components/apiKeys/ApiKeysCopyPopup.vue';

import { ApiKeysApiGql } from '@/api/apiKeys';
import { makeApiKeysModule } from '@/store/modules/apiKeys';
import { makeNotificationsModule } from '@/store/modules/notifications';
import { NotificatorPlugin } from '@/utils/plugins/notificator';
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();
localVue.use(Vuex);
const notificationPlugin = new NotificatorPlugin();
localVue.use(notificationPlugin);
const apiKeysApi = new ApiKeysApiGql();
const apiKeysModule = makeApiKeysModule(apiKeysApi);
const notificationsModule = makeNotificationsModule();

const store = new Vuex.Store({ modules: { notificationsModule, apiKeysModule }});

describe('ApiKeysCopyPopup', () => {
    it('renders correctly', () => {
        const wrapper = mount(ApiKeysCopyPopup, {
            store,
            localVue,
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('function onCloseClick works correctly', () => {
        const wrapper = mount(ApiKeysCopyPopup, {
            store,
            localVue,
        });

        wrapper.vm.onCloseClick();

        expect(wrapper.vm.$data.isCopiedButtonShown).toBe(false);
        expect(wrapper.emitted()).toEqual({'closePopup': [[]]});
    });

    it('function onCopyClick works correctly', async () => {
        const wrapper = mount(ApiKeysCopyPopup, {
            store,
            localVue,
        });

        await wrapper.vm.onCopyClick();

        expect(wrapper.vm.$data.isCopiedButtonShown).toBe(true);
    });
});
