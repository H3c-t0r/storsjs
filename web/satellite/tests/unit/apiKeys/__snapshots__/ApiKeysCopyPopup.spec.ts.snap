// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiKeysCopyPopup renders correctly 1`] = ``;
